src/gui/pointcloud/qgspointcloudrendererpropertieswidget.cpp

```diff
@@ -189,6 +201,20 @@ void QgsPointCloudRendererPropertiesWidget::syncToLayer( QgsMapLayer *layer )
     mHorizontalTriangleCheckBox->setChecked( mLayer->renderer()->horizontalTriangleFilter() );
     mHorizontalTriangleThresholdSpinBox->setValue( mLayer->renderer()->horizontalTriangleFilterThreshold() );
     mHorizontalTriangleUnitWidget->setUnit( mLayer->renderer()->horizontalTriangleFilterUnit() );
+
+    if ( !mLayer->dataProvider()->subIndexes().isEmpty() )
+    {
+      mLabels->setChecked( mLayer->renderer()->showLabels() );
+      // set some better defaults for label format
+      QgsTextFormat textFormat = mLayer->renderer()->labelTextFormat();
+      QgsTextBufferSettings settings;
+      settings.setEnabled( true );
```

Comment:
Users are always able to change all the text settings. 
These options should just represent some reasonable options which are set when the layer is loaded. 

I changed the default textFormat to use `QgsStyle::defaultStyle()->defaultTextFormat()`. However I think it's better to keep the settings here instead of `QgsPointCloudLayerRenderer` constructor as the preset is more oriented towards the text format widget which is the source of truth not the renderer.

Reply:
> Users are always able to change all the text settings.

That's not the case -- as soon as they open the layer in this widget, it will reset the format to apply a buffer, even if they'd previously unchecked that!

In any case it's bad practice to set defaults in a widget -- the behavior will be fragile, and eg won't apply if you're creating layers directly via API from a headless environment.

Reply:
Huh okay, I think I understand what you mean now. It's funny that I tried it now and the setting gets reapplied as it is visible in the text format widget, however the actual rendered labels don't change :)